### ISSUE TITLE
Debugger script file;  File Not Found

### ISSUE THREAD

**Issue by TommyGH** [AppleWin/AppleWin#463]
This is the error message that started my confusion:

Couldn't load filename: C:\Users\TOM\Desktop\WorkFolder\SubFolder

Things to think about when looking at my instructions to reproduce:

A. It would be nice to see the filename in the error message.
B. I was really confused at step 16 that I could not enter just the filename to execute the script.
C. The help file was useless in trying to pinpoint my problem.
D. Where is the default path stored?  When does it change?  How do I change it?

The following instructions are actually a subset of some of the things I was trying before I actually "solved" my problem.

I have WorkFolder on my Windows desktop with the following:

- Shortcut to AppleWin.exe
- Disk1.dsk
- debug.txt
- SubFolder with Disk2.dsk

Instructions:

1. Start AppleWin
2. Insert Disk1
3. F7
4. RUN debug.txt (no problem)
5. Insert Disk2
6. RUN debug.txt (no problem)
7. Close AppleWin
8. Start AppleWin
9. F7
10. RUN debug.txt

Couldn't load filename: C:\Users\TOM\Desktop\WorkFolder\SubFolder

No filename was on the line, just the path.  This confused me.  Was the problem with debug.txt? Or was it with Disk2?  Or A2_USER1.SYM? Or something else?

I notice Disk2 in Drive1 was from SubFolder.
11. Eject Disk2

12. RUN debug.txt
Couldn't load filename: C:\Users\TOM\Desktop\WorkFolder\SubFolder

13. Insert Disk1.dsk
14. RUN debug.txt
Couldn't load filename: C:\Users\TOM\Desktop\WorkFolder\SubFolder

15. RUN "C:\Users\TOM\Desktop\WorkFolder\SubFolder\debug.txt"
Debug.txt runs properly.

16. RUN debug.txt
Couldn't load filename: C:\Users\TOM\Desktop\WorkFolder\SubFolder

At this point I've got Disk1 in drive1, I've successfully executed debug.txt once (by using the full path) and I don't see what else to change so that I can just type "debug.txt" like I did at steps 4 and 6.

17. Close AppleWin
18. Start AppleWin
Disk1 is in Drive1.

19. F7
20. RUN debug.txt

Success.  Now I can use just the filename.

**Comment by Michaelangel007** [comment]
> A. It would be nice to see the filename in the error message.

Agreed!  There is absolutely no excuse for the filename not to be displayed. The filename is probably getting truncated somewhere along the lines.

The cause of this bug is that `RUN` _requires_ the filename to be double-quoted.   Same for `BSAVE` and `BLOAD`, and all filenames in general inside the debugger.  (This was a decision to make the debugger command line parsing easier.)

Hmm, I see there is no `HELP RUN` like there is for `HELP BSAVE`.

The TL:DR; is that there are 2 bugs:

* RUN foo, without double-quotes is not showing the filename
* HELP RUN is incomplete -- it needs syntax and examples.

**Comment by TommyGH** [comment]
> The cause of this bug is that  RUN  requires the filename to be double-quoted.

This statement has me a little confused.  Currently I am able to **usually** execute the run command without using any quotes:

RUN debug.txt

Just to be clear, the problem I encountered happened after I switched boot disks,  closed AppleWin, and then restarted AppleWin.

Also, the CHM help does show an example using quotation marks:
  "RUN text.txt"
It also says, "the filename **may** be quoted".   My emphasis to the word "may".

**Comment by TommyGH** [comment]
Ok, I've just re-discovered the PWD command, and it sheds more light.

When I start AppleWin with a disk already in the drive, PWD shows the path to the directory where that disk image is located.  If my debug.txt file is in that directory, I can get by with just RUN DEBUG.TXT.

If I keep AppleWin open, but load a disk image from another directory on my computer, then PWD is not updated, and again RUN DEBUG.TXT works just fine.

Now when I close AppleWin, and then restart AppleWin, PWD shows the path to the **second** disk image which is in a different directory from my DEBUG.TXT.  And therefore when I enter RUN DEBUG.TXT it fails.

I guess some of my confusion could be attributed to user error, but maybe better descriptions in the help file would have helped.  Perhaps the "Scripts" section in the CHM file should mention that the script filename by itself uses the default path (displayed using PWD) which can be changed with the CD command.

**Comment by TommyGH** [comment]
Two more similar open issues:
#111  Debugger RUN <script-file> still not 100%
#114 Debugger: symuser load filename.sym fails

The use of CD to change paths in AppleWin has me terribly confused (or is broken).

1. Start AppleWin
2. F7
3. PWD
  C:\Users\TOM\Desktop\Conflict In Vietnam\extra files\
4. CD ..
 C:\Users\TOM\Desktop\Conflict In Vietnam\extra files\..\
5. CD "extra files"
  C:\Users\TOM\Desktop\Conflict In Vietnam\extra files\..\extra files\

There's not much help for CD in AppleWin.

**Comment by tomcw** [comment]
Related in 75f6ab9 I changed the `run` command to put the 'file not found' msg & pathnames on different lines, eg:
```
Couldn't load filename:
<pathname>
```
This gives 24 chars of visibility for the pathname.
